flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcReaders.java

```diff
@@ -246,7 +256,7 @@ public void setBatchContext(long batchOffsetInFile) {
 
     StructReader(List<OrcValueReader<?>> readers, Types.StructType struct, Map<Integer, ?> idToConstant) {
       super(readers, struct, idToConstant);
-      this.numFields = readers.size();
+      this.numFields = struct.fields().size();
```

Comment:
This fixes Orc Reader (with partition) bug.

Reply:
Nice catch,  if the schema is a projected read schema,  then the `numFields` will be mismatched.